app/code/Magento/MediaGalleryApi/Model/DataExtractorInterface.php

```diff
@@ -0,0 +1,23 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGalleryApi\Model;
+
+/**
+ * Extract data from an object using available getters
+ */
+interface DataExtractorInterface
```

Comment:
Why does the extractor need to be part of public API? It looks more like an object internal to the module.

Reply:
It used by multiple modules related to Adobe Stock API integration.

Reply:
The extractor interface was created to use the extractor in other modules (Adobe Stock Integration) as every Interface that does not include getData method needs this untility